src/azure-cli/azure/cli/command_modules/appconfig/_snapshot_custom_client.py

```diff
@@ -0,0 +1,540 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long, too-many-locals, too-many-statements, too-many-branches, too-few-public-methods
+
+from azure.appconfiguration import AzureAppConfigurationClient, ConfigurationSetting
+from azure.appconfiguration._generated.models import Error as AppConfigError
+from enum import Enum
+from azure.core.rest import HttpRequest
+from azure.core.paging import ItemPaged
+from azure.core.utils import case_insensitive_dict
+from azure.core.exceptions import ClientAuthenticationError, ResourceExistsError, ResourceNotFoundError, HttpResponseError, ResourceModifiedError, map_error
+from msrest import Serializer
+from typing import Dict, List, Optional, Any
+import json
+
+from ._constants import SnapshotConstants
+from ._snapshotmodels import Snapshot, SnapshotListResult
+
+
+class ProvisioningStatus:
+    PROVISIONING = "provisioning"
+    READY = "ready"
+    ARCHIVED = "archived"
+    FAILED = "failed"
+
+
+class RequestMethod:
+    GET = "GET"
+    PUT = "PUT"
+    PATCH = "PATCH"
+    DELETE = "DELETE"
+
+
+_ERROR_MAP = {
+    401: ClientAuthenticationError,
+    404: ResourceNotFoundError,
+    409: ResourceExistsError,
+    412: ResourceModifiedError
+}
+
+
+def _build_request(
+    template_url_default,
+    method,
+    *,
+    path_arguments={},
+    if_match=None,
+    if_none_match=None,
+    sync_token=None,
+    **kwargs
+):
+    _headers = case_insensitive_dict(kwargs.pop("headers", {}) or {})
+    _params = case_insensitive_dict(kwargs.pop("params", {}) or {})
+
+    api_version = kwargs.pop("api_version", SnapshotConstants.API_VERSION)
+    accept = _headers.pop(
+        "Accept", "application/vnd.microsoft.appconfig.keyset+json, application/json, application/problem+json"
```

Comment:
`application/vnd.microsoft.appconfig.keyset+json` that should only be for the '/keys' endpoint.

"application/json, application/problem+json" should be enough for this purposes.

Reply:
This has been updated, thanks!